truss/templates/shared/lazy_data_resolver.py

```diff
@@ -0,0 +1,53 @@
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List
+
+import pydantic
+import yaml
+from truss.util.download import download_from_url_using_requests
+
+LAZY_DATA_RESOLVER_PATH = Path("/bptr/bptr-manifest")
+
+
+class Resolution(pydantic.BaseModel):
+    url: str
+    expiry: datetime
+
+
+class BasetenPointer(pydantic.BaseModel):
+    """Specification for lazy data resolution for download of large files, similar to Git LFS pointers"""
+
+    resolution: Resolution
+    uid: str
+    file_name: str
+    hashtype: str
+    hash: str
+    size: int
+
+
+class BasetenPointerManifest(pydantic.BaseModel):
+    pointers: List[BasetenPointer]
+
+
+class LazyDataResolver:
+    def __init__(self, data_dir: Path):
+        self._data_dir: Path = data_dir
+        self._bptr_resolution: Dict[str, str] = _read_bptr_resolution()
+
+    def fetch(self):
+        for file_name, resolved_url in self._bptr_resolution.items():
+            download_from_url_using_requests(resolved_url, self._data_dir / file_name)
```

Comment:
Should we use b10cp here with `_download_from_url_using_b10cp`, or the b10cp proxy container will work if we use requests?

Reply:
yup the idea was that b10cp would proxy this traffic – let me test that this works

Reply:
Think this will be best to test e2e when we have the manifest plumbing to the service – will wait until then, we should pretty easily be able to switch the download mechanism in any case